package/MDAnalysis/core/groups.py

```diff
@@ -2554,19 +2554,46 @@ def write(self, filename=None, file_format="PDB",
             ``"conect"``: write only the CONECT records defined in the original
             file. ``"all"``: write out all bonds, both the original defined and
             those guessed by MDAnalysis. ``None``: do not write out bonds.
-            Default os ``"conect"``.
+            Default is ``"conect"``.
+        frames:
+            An ensemble of frames to write. The ensemble can be an list or
+            array of frame indices, a mask of booleans, an instance of
+            :class:`slice`, or an indexed trajectory. By default, 'frames' is
+            set to ``None`` and only the current frame is written.
 
 
         .. versionchanged:: 0.9.0 Merged with write_selection. This method can
             now write both selections out.
+        .. versionchanged:: 0.19.0
+            Can write multiframe trajectories with the 'frames' argument.
         """
         # check that AtomGroup actually has any atoms (Issue #434)
         if len(self.atoms) == 0:
             raise IndexError("Cannot write an AtomGroup with 0 atoms")
 
         trj = self.universe.trajectory  # unified trajectory API
-
-        if trj.n_frames == 1:
+        if frames is None:
+            trj_frames = trj[::]
+        else:
+            try:
+                test_trajectory = frames.trajectory
+            except AttributeError:
+                trj_frames = trj[frames]
+            else:
+                if test_trajectory is not trj:
+                    raise ValueError(
+                        'The trajectory of {} provided to the frames keyword '
+                        'attribute is different from the trajectory of the '
+                        'AtomGroup.'.format(frames)
+                    )
+                trj_frames = frames
+            if len(trj_frames) > 1 and kwargs.get("multiframe") == False:
```

Comment:
If we do something like the `frame=True` suggested above, then we can deprecate `multiframe` for 1.0 and just make it mean `frames=True` in the interim.

Reply:
The `multiframe` argument is only passed to the writer; the gymnastic in `write` is actually useless so I removed it. I kept that test, though as the case could still happens and makes no sense.